File: src/main/java/com/dodecaedro/library/pojo/Book.java
package com.dodecaedro.library.pojo;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.time.ZonedDateTime;
import java.util.List;

@Data
@EqualsAndHashCode(of = {"isbn"})
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "book")
public class Book implements Serializable {
  @Id
  @NotNull
  @Column(name = "id")
  @GeneratedValue
  private Long bookId;

  @NotNull @Column private String title;

  @NotNull @Column private String isbn;

  @NotNull
  @Column(name = "bought_date")
  private ZonedDateTime dateTimeBought;

  @OneToMany(cascade = CascadeType.ALL, mappedBy = "book")
  private List<Borrow> borrows;

  @PrePersist
  void createdAt() {
    if (this.dateTimeBought == null) {
      this.dateTimeBought = ZonedDateTime.now();
    }
  }
}

File: library-model/src/main/kotlin/com/dodecaedro/library/domain/model/Book.kt
package com.dodecaedro.library.domain.model

import java.time.ZonedDateTime
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Book(
	@field:Id val id: UUID? = UUID.randomUUID(),
	var title: String?,
	var isbn: String?,
	@field:Column(name = "bought_date")
	var dateTimeBought: ZonedDateTime?,
)
